### ISSUE TITLE
A11y traversal order incorrect for slivers hidden behind trailing edge

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#17023]
See https://github.com/flutter/flutter/pull/17021/files#diff-8cdfa193beff118de7c340e5afbc7490R324 for a test that currently fails.

This is only an issue if you have multiple slivers (e.g. multiple SliverToBoxAdapters). It is explicitly not an issue for a single lazy sliver (like SliverList, SliverGrid, etc).

/cc @yjbanov
